### ISSUE TITLE
Propose new legislation

### ISSUE THREAD

**Issue by dsernst** [unitedvote/united.vote#1]
### Version 1
1. **Any verified user can propose new legislation**. Simple, plain text input.
2. This creates a unique Proposed Legislation page for it. **Any verified user can vote on it** and leave comments, same as Introduced Legislation.
3. If Proposed Legislation receives **liquid Yea's greater than 50% of the verified** people in that district, its proposal is considered accepted by the liquid district.
4. The item moves to a separate list: Liquid Majority Wants Introduced. The elected legislator is notified. **Until they introduce it, its liquid Yea's are now scored as `against_constituents`** and liquid Nay's as `with_constituents` for that legislator.
5. **The legislator is invited to notify the platform when it's been introduced** in the elected legislature. If they introduce it themselves, their scores for the item reverse: `yeas = with_constituents`, `nays = against_constituents`.

**Comment by cehsu** [comment]
Is linking out to https://lawmaker.io of interest here?

**Comment by dsernst** [comment]
Don't think so right now, for a few reasons but perhaps most of all because they're CA only, whereas we're Congress only.

**Comment by cehsu** [comment]
Thanks for the context @dsernst!

**Comment by dsernst** [comment]
I began work on this in the [`propose-legislation` branch](https://github.com/unitedvote/united.vote/commit/8520e754b612b9d607ed5a513e79d7dc3fc97a47).

---------

- [x] Adds a `New` button to the legislation list to take you to `/legslation/new`.

<img width="511" alt="screen shot 2018-04-24 at 2 06 09 am" src="https://user-images.githubusercontent.com/6340841/39177416-1e444a2c-4764-11e8-84f2-0c66c67d90c0.png">

--------

- [x] `/legislation/new` is a simple form to enter a title and description.

![screen shot 2018-04-24 at 2 07 35 am](https://user-images.githubusercontent.com/6340841/39177489-4c95c220-4764-11e8-8ed3-fd98b2719b24.png)

- [x] After you press save, you're taken to `/legislation/new/yours`, where you can see a list of all the items you've proposed. They're all [listed as `Unpublished`](https://github.com/unitedvote/united.vote/commit/8520e754b612b9d607ed5a513e79d7dc3fc97a47#diff-d4f14ee4f966c83d6946d860ba00d960R49) at first, on the idea that you can continue to revise them before choosing to make public.

![screen shot 2018-04-24 at 2 09 12 am](https://user-images.githubusercontent.com/6340841/39177612-9b9ea77e-4764-11e8-8eac-25dfa39cacbb.png)

- [x] You can click into any of your items to go to it's dedicated page `/:username/:proposalId`
  -  [ ] Not much to do on this page yet
  - Currently only you can view it
  - URL is namespaced to you. `:proposalId` is a standard UUID. The thinking is you'd be able to assign a `shortname` as an easier url, so it'd be like github repo namespacing.

![screen shot 2018-04-24 at 2 10 43 am](https://user-images.githubusercontent.com/6340841/39177657-b8855ebe-4764-11e8-8a00-72752d45b116.png)

### Todo:

- [ ] Need to add a way to edit your draft legislation
- [ ] Need to be able to publish your legislation publicly.
- [ ] Need to allow other people to be able to vote and comments For/Against your proposed legislation.

**Comment by alexmingoia** [comment]
I'm starting to digest this... I have a bunch of questions on your outline:

1. "new" can be confused with recent legislation, because it's both a noun and a verb. Can we use "propose" in the URL instead (`/legislation/propose` instead of `/legislation/new`)?
2. Can people propose legislation anonymously? The URL `/:username/:proposal_id` somewhat implies that they cannot. Personally, I don't want a public profile page but I do want to propose legislation. In which case, using the username in the URL is somewhat problematic. What about using `/legislation/proposed/:id` instead (or with the short name `/legislation/proposed/:short_name`)? Also, I would think it'd be better to have proposed legislation on the public profile (`/:username`) rather than a separate page.
3. What's the thinking behind requiring proposed legislation to be per-district? I ask because I imagined that proposed legislation would have nothing to do with districts. It doesn't matter which legislator introduces legislation, since the whole legislature votes on it and legislation applies to the whole US. So many thoughts here... Shouldn't any legislator be able to introduce proposed legislation? Doesn't requiring a quorum of the district mean that the district an author happens to live in will greatly affect its succes? Liberal legislation will be difficult to introduce in conservative districts, and vice-versa.
4. Is there quorum requirements for proposed legislation, and when is proposed legislation voting considered "closed" and passed? There's currently no rollcall voting or voting windows on the platform. For existing legislation, this is kind of works because the vote can be considered "closed" once the legislature actually takes a vote. But with proposed legislation this becomes more confusing. When does the quorum and majority requirements actually take affect? If one day the proposed legislation is 51% yea / 49% nay, and then the next day it's 90% nay, is it no longer recommended for introduction?
5. The great thing about the "with constituents / against constituents" is that it's straightforward and easy to understand. I'd be disappointed to include the proposed legislation in this, because then it ceases to be an obvious measure of their representation.

I saw you started work on both the front-end and the API. Let's leave the API for after the front-end is done (use dummy data), because it's not clear how everything will work and what an effective database schema looks like.
